csharp.test/TestLogicalTypeRoundtrip.cs

```diff
@@ -143,6 +143,68 @@ public static void TestDateTimeRoundTrip(DateTimeKind kind, TimeUnit timeUnit)
             Assert.AreEqual(kind, kinds.First());
         }
 
+        [Test]
+        [NonParallelizable]
+        public static void TestAppSwitchDateTimeKindUnspecified()
+        {
+            AppContext.SetSwitch("ParquetSharp.UseDateTimeKindUnspecified", true);
```

Comment:
Would that not be bad practice, as you start exposing implementation details where its otherwise unnecessary?

Also this is inline with how ASP.NET Core does it: `{namespace}.{effect}`:
  * https://github.com/dotnet/aspnetcore/blob/5026aefddd69322da633d4b29a3d24c4df0b51b9/src/Http/Http/src/Internal/ResponseCookies.cs#L17
  * https://github.com/dotnet/aspnetcore/blob/84c0baa9a57bd6448cd1bf78b64d2c1658bf5ef6/src/Identity/Core/src/SignInManager.cs#L387

Reply:
I would be fine with `ParquetSharp.ReadDateTimeKindAsUnspecified`

Reply:
👍 `ParquetSharp.ReadDateTimeKindAsUnspecified` sounds good to me